be/src/formats/parquet/file_reader.cpp

```diff
@@ -59,7 +59,7 @@ Status FileReader::init(HdfsScannerContext* ctx) {
     // set existed SlotDescriptor in this parquet file
     std::unordered_set<std::string> names;
     _meta_helper->set_existed_column_names(&names);
-    _scanner_ctx->set_columns_from_file(names);
+    _scanner_ctx->update_materialized_columns(names);
```

Comment:
could you explain how it works for reader?

Reply:
Suppose in hive table, we have table of three fields like a, b, c.

But in parquet file, there are only two fields: a, b.

So when we `select a,b,c from t`, c should be handled specially, because it does not exist in parquet file. We can mark it as non-existed column.

And that's the purpose of this function `update_materialized_columns`: Mark out some non-existed columns by comparing columns from plan and columns from file.